x-pack/plugins/apm/server/lib/services/get_service_transaction_groups_statistics.ts

```diff
@@ -0,0 +1,179 @@
+  );
+
+  const response = await apmEventClient.search({
+    apm: {
+      events: [
+        getProcessorEventForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
+            { term: { [TRANSACTION_TYPE]: transactionType } },
+            { range: rangeFilter(start, end) },
+            ...getDocumentTypeFilterForAggregatedTransactions(
+              searchAggregatedTransactions
+            ),
+            ...esFilter,
+          ],
+        },
+      },
+      aggs: {
+        total_duration: { sum: { field: TRANSACTION_DURATION } },
+        transaction_groups: {
+          terms: {
+            field: TRANSACTION_NAME,
+            include: transactionNames,
+            size: transactionNames.length,
+          },
+          aggs: {
+            transaction_group_total_duration: {
+              sum: { field: TRANSACTION_DURATION },
+            },
+            timeseries: {
+              date_histogram: {
+                field: '@timestamp',
+                fixed_interval: intervalString,
+                min_doc_count: 0,
+                extended_bounds: {
+                  min: start,
+                  max: end,
+                },
+              },
+              aggs: {
+                throughput_rate: {
+                  rate: {
+                    field: TRANSACTION_DURATION,
+                    unit: 'minute',
+                    mode: 'value_count',
+                  },
+                },
+                ...getLatencyAggregation(latencyAggregationType, field),
+                [EVENT_OUTCOME]: {
+                  filter: {
+                    term: { [EVENT_OUTCOME]: EventOutcome.failure },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const buckets = response.aggregations?.transaction_groups.buckets ?? [];
+
+  const totalDuration = response.aggregations?.total_duration.value;
+
+  return buckets.reduce((acc, bucket) => {
+    const transactionName = bucket.key;
+
+    const latency = bucket.timeseries.buckets.map((timeseriesBucket) => ({
+      x: timeseriesBucket.key,
+      y: getLatencyValue({
+        latencyAggregationType,
+        aggregation: timeseriesBucket.latency,
+      }),
+    }));
+
+    const throughput = bucket.timeseries.buckets.map((timeseriesBucket) => ({
+      x: timeseriesBucket.key,
+      y:
+        timeseriesBucket.throughput_rate.value !== null
+          ? timeseriesBucket.throughput_rate.value / 100
+          : null,
+    }));
+
+    const errorRate = bucket.timeseries.buckets.map((timeseriesBucket) => ({
+      x: timeseriesBucket.key,
+      y:
+        timeseriesBucket.doc_count > 0
+          ? timeseriesBucket[EVENT_OUTCOME].doc_count /
+            timeseriesBucket.doc_count
+          : null,
```

Comment:
I believe this is incorrect if we don't filter for EVENT_OUTCOME being either 'success' or 'failure', as it will include 'unknown' in the total count. Looks like that bug already existed but good opportunity to fix.